src/flb_utils.c

```diff
@@ -1014,16 +1014,30 @@ int flb_utils_write_str_buf(const char *str, size_t str_len, char **out, size_t
     return 0;
 }
 
-static char *flb_copy_host(const char *string, int pos_init, int pos_end)
+char *flb_copy_host(const char *string, int pos_init, int pos_end)
 {
     if (string[pos_init] == '[') {            /* IPv6 */
-        if (string[pos_end-1] != ']')
+        if (string[pos_end-1] != ']') {
             return NULL;
-
+        }
         return mk_string_copy_substr(string, pos_init + 1, pos_end - 1);
     }
-    else
+    else {
         return mk_string_copy_substr(string, pos_init, pos_end);
+    }
+}
+
+char *flb_copy_host_sds(const char *string, int pos_init, int pos_end)
```

Comment:
if this interface will be from public use (non static), the name must be prefixed according the file, e.g: flb_utils_host_copy_sds()